demos/pkcs12/pkread.c

```diff
@@ -48,56 +48,64 @@ static char *find_friendly_name(PKCS12 *p12)
 int main(int argc, char **argv)
 {
     FILE *fp;
-    EVP_PKEY *pkey;
-    X509 *cert;
+    EVP_PKEY *pkey = NULL;
+    X509 *cert = NULL;
     STACK_OF(X509) *ca = NULL;
-    PKCS12 *p12;
-    const char *name;
-    int i;
+    PKCS12 *p12 = NULL;
+    char *name;
```

Comment:
Should this be set to NULL? (`char *name = NULL;`)

Reply:
It seems the `OPENSSL_free` in the `err` section could free this pointer before setting a value to it.

Reply:
It look this way to me too